### PR TITLE
Include test "compilation" logic

### PR DESCRIPTION
**Pull request by jugglinmike**
Hey @rwaldron,

Since we started, we've been expected to incorporate the source code from the `test262-compiler` project directly. The feature request in gh-2 kind of forces this issue because it describes disabling some (but not all) of the functionality that module provides.

One additional benefit (as noted in the first commit on this branch) is that this reduces the total number of Node.js dependencies by close to 50%.

But more importantly: while implementing this, I found that `test262-compiler` violates the Test262 "interpreting" guidelines in a number of ways. In the interest of transparency (so you can see exactly what I modified after importing the code), I've addressed those violations in dedicated commits. I wanted to call this out because it's possible that those inconsistencies are actually required by `test262-harness`. If that's the case, then removing them might make it difficult to use `test262-stream` from `test262-harness`. I don't believe this is the case, but I'd appreciate it if you could verify.

Thanks!

### CODE REVIEW

**Comment by rwaldron** [comment]
First of all, I'm sorry for dropping the ball on reviewing this. Secondly, I wonder if removing `yargs` dependency from `test262-compiler` is a safer approach? I'm concerned about keeping this project's `compiler` and `test262-compiler` in sync. Alternatively, we could make `test262-compiler` _just_ the compiler, and create a new thing that is the CLI tool?

**Comment by jugglinmike** [comment]
I agree that we could also reduce this project's dependencies by modifying the `tets262-compiler` project, but the dependency removal was an unintended benefit.

I don't think the "stream"/"compiler" separation is useful, and I expect that maintaining it would just involve making more work for ourselves. [npm only lists three dependent projects for `test262-compiler`](https://www.npmjs.com/package/test262-compiler), so if we intend to use `test262-stream` inside of `test262-harness`, then we could probably deprecate `test262-compiler` altogether.

**Comment by rwaldron** [comment]
@jugglinmike thanks for clarifying these points for me—I'm on board.

File: lib/compilerdeps.js [hunk 1]
```diff
@@ -28,7 +28,3 @@ function $DONE(err) {
   $262.destroy();
 }
 
-function $LOG(str) {
-  print(str);
-}
```
Comment (rwaldron) [inline lib/compilerdeps.js]: I've confirmed that this is not used an any existing test/* or harness/* files in test262

**Comment by rwaldron** [comment]
Is there any special handling that you want for merging this?

**Comment by jugglinmike** [comment]
I'd like to see the commits I've submitted landed without squashing since they both import code authored by someone else and then introduce my own modifications. It's your call whether to use a merge commit or not.

(Sorry for the delay, I missed the e-mail notifications for your latest comments.)

**Comment by rwaldron** [comment]
Thanks @jugglinmike

**Pull request merged**